src/utilities/tuples.ts

```diff
@@ -0,0 +1,16 @@
+export type TupleType<T extends string[]> = {
+  tuple: T,
+  isTuple: (value: unknown) => value is T,
+}
+
+export function createTuple<const T extends string[]>(tuple: T): TupleType<T> {
```

Comment:
as-is this would only support tuples of strings? not other types nor variadic tuples?

Reply:
I hadn't thought about making this more general. Its design is to simplify the creating of string tuple types. Though it could work with any value. See https://github.com/PrefectHQ/prefect-ui-library/pull/1982/commits/b52f531bac1cf6570e5429155e6abf2178eeeeca